src/lean_spec/subspecs/containers/vote/vote.py

```diff
@@ -0,0 +1,89 @@
+"""Attestation-related container definitions."""
+
+from lean_spec.subspecs.containers.slot import Slot
+from lean_spec.types import Bytes4000, Container, Uint64
+
+from ..checkpoint import Checkpoint
+from .types import AggregatedSignatures, AggregationBits
+
+
+class AttestationData(Container):
+    """Attestation content describing the validator's observed chain view."""
+
+    slot: Slot
+    head: Checkpoint
+    target: Checkpoint
+    source: Checkpoint
+
+
+class Attestation(Container):
+    """Validator specific attestation wrapping shared attestation data."""
+
+    validator_id: Uint64
+    data: AttestationData
+
+    @property
+    def slot(self) -> Slot:
+        """Return the attested slot."""
+        return self.data.slot
+
+    @property
+    def head(self) -> Checkpoint:
+        """Return the attested head checkpoint."""
+        return self.data.head
+
+    @property
+    def target(self) -> Checkpoint:
+        """Return the attested target checkpoint."""
+        return self.data.target
+
+    @property
+    def source(self) -> Checkpoint:
+        """Return the attested source checkpoint."""
+        return self.data.source
+
+
+class SignedAttestation(Container):
+    """Validator attestation bundled with its signature."""
+
+    message: Attestation
+    signature: Bytes4000
+    """Signature produced by the lean signature VM.
+
+    Unlike BLS, signatures over ValidatorAttestation messages that share the
+    same underlying AttestationData can be aggregated efficiently by the lean
+    signature VM.
+    """
```

Comment:
We don't need to be more specific, otherwise this is confusing.

```suggestion
    """Signature aggregation produced by the leanVM (SNARKs in the future)."""
```